### ISSUE TITLE
New INJECT_FACTS_AS_VARS deprecation in 2.20 is not reported by linter

### ISSUE THREAD

**Issue by marcelmamula** [ansible/ansible-lint#4857]
##### Summary

New ansible-core 2.20 brought deprecation of using ansible facts without dictionary, but linter does not show it in warning, making it difficult to find it.

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
ansible-lint 25.11.1 using ansible-core:2.20.0 ansible-compat:25.11.0 ruamel-yaml:0.18.16 ruamel-yaml-clib:0.2.15
```

- ansible installation method: pip3 inside of python 3.13 VENV
- ansible-lint installation method: pip3 inside of python 3.13 VENV

##### STEPS TO REPRODUCE

Execute linter for small playbook using `ansible_hostname`.
```console (paste below)
    - name: Test ansible_hostname
      ansible.builtin.debug:
        msg: "{{ ansible_hostname}}"
```

##### Desired Behavior

Show warning in linter output, same like ansible-playbook generates.
```bash
[WARNING]: Deprecation warnings can be disabled by setting `deprecation_warnings=False` in ansible.cfg.
[DEPRECATION WARNING]: INJECT_FACTS_AS_VARS default to `True` is deprecated, top-level facts will not be auto injected after the change. This feature will be removed from ansible-core version 2.24.
Origin: /home/mmamula/workspace/tests/debug.yml:23:14

21     - name: Test ansible_hostname
22       ansible.builtin.debug:
23         msg: "{{ ansible_hostname }}"
                ^ column 14

Use `ansible_facts["fact_name"]` (no `ansible_` prefix) instead.
```

##### Actual Behavior

Linter does not show any warnings for this tiny playbook using ansible_hostname.

```paste below
Passed: 0 failure(s), 0 warning(s) in 1 files processed of 1 encountered.
```

[minimum complete verifiable example]: http://stackoverflow.com/help/mcve

**Comment by nre-ableton** [comment]
+1, I would really like to see ansible-lint warn for `ansible_*` facts that are available in the `ansible_facts` dict. However, it's important to note that a number of `ansible_*` facts are _not_ exposed in this dict, including:

- `ansible_become_pass`
- `ansible_check_mode`
- `ansible_config_file`
- `ansible_diff_mode`
- `ansible_forks`
- `ansible_group`
- `ansible_host`
- `ansible_inventory_sources`
- `ansible_local`
- `ansible_playbook_python`
- `ansible_run_tags`
- `ansible_skip_tags`
- `ansible_user`
- `ansible_verbosity`
- `ansible_version`

**Comment by ssbarnea** [comment]
Seems legit, PR welcome!

**Comment by nre-ableton** [comment]
> Seems legit, PR welcome!

@ssbarnea I'd be happy to do the work, but I'm not at all familiar with the codebase. Is there a starting point function/class/etc that I should look at to base my work on?

**Comment by Jkhall81** [comment]
@nre-ableton Hello.  Here is a quick 'roadmap'.

- `src/ansiblelint/constants.py`: Define a "safe list" (internal vars vs. facts).
- `src/ansiblelint/rules/`: In here you would need a `.py` and `.md` file.  This is where the rule logic and documentation / code examples will live.
-  `test/`: Pytest tests to make sure everything works.

**Comment by nre-ableton** [comment]
Thanks @Jkhall81! I'm not sure when I'll have time to look at implementing this, but in the meantime, if someone else sees this comment, they would also have a good starting point.

**Comment by ssbarnea** [comment]
Personal hint: feel free to make use of AI to do it, you will be surprised how little code you might have to type.

**Comment by ariel-anieli** [comment]
> Thanks [@Jkhall81](https://github.com/Jkhall81)! I'm not sure when I'll have time to look at implementing this, but in the meantime, if someone else sees this comment, they would also have a good starting point.

@nre-ableton, @Jkhall81; I'll work on this. Thanks for the roadmap.
